src/util/common.jl

```diff
@@ -0,0 +1,308 @@
+
+"updates the solve time limit of an optimizer"
+function _update_optimizer_time_limit!(optimizer::MathOptInterface.OptimizerWithAttributes, time_limit::Real)
+        new_params = typeof((optimizer.params))()
+    for param in optimizer.params
+        if param[1] != MathOptInterface.TimeLimitSec()
+            push!(new_params, param)
+        end
+    end
+    empty!(optimizer.params)
+    for param in new_params
+        push!(optimizer.params, param)
+    end
+    push!(optimizer.params, (MathOptInterface.TimeLimitSec()=>time_limit))
+
+    return optimizer
+end
+
+
+"""
+        `_fill_missing_variables!(sol::Dict{String,<:Any}, network::Dict{String,<:Any})`
+    add network values for components removed for status=0 from solution
+"""
+function _fill_missing_variables!(sol::Dict{String,<:Any}, network::Dict{String,<:Any})
+
+    # Is there a way to do this based on formulation, any default comps?
+    variable_fill = Dict(
+        "bus"=>Dict{String,Any}("va"=>0.0,"status"=>0, "bus_type"=>4, "vm"=>NaN),
+        "gen"=>Dict{String,Any}("gen_status"=>0,"pg"=>0.0, "qg"=>NaN),
+        "branch"=>Dict{String,Any}("qf"=>NaN,"qt"=>NaN,"br_status"=>0,"pt"=>0.0,"pf"=>0.0),
+        "dcline"=>Dict{String,Any}("qf"=>NaN,"qt"=>NaN,"br_status"=>0,"pt"=>0.0,"pf"=>0.0),
+        "load"=>Dict{String,Any}("status"=>0.0,"qd"=>0.0,"pd"=>0.0),
+        "shunt"=>Dict{String,Any}("status"=>0, "gs"=>0.0, "bs"=>0.0),
+        #"storage"=>Dict{String,Any}("status"=>0) verify storage is working
+    )
+
+    for (nw_id, sol_net) in sol["solution"]["nw"]
+        for (comp_type,comp_vars) in variable_fill
+            for (comp_id,comp_net) in network[comp_type]
+                if !(haskey(sol_net,comp_type))
+                    sol_net[comp_type] = Dict{String,Any}()
+                end
+                if !(haskey(sol_net[comp_type], comp_id))
+                    sol_net[comp_type][comp_id]=comp_vars
+                end
+            end
+        end
+    end
+end
+
+
+"get repairs in each period of the solution data"
+function _get_component_repairs(solution)
+    if haskey(solution,"solution")
+        network = solution["solution"]
+    else
+        network = solution
+    end
```

Comment:
My impression is that this will encourage sloppy code that calls this function.  I think it would be better if the callers of this function do the `solution["solution"]` step.